packages/core/src/indexing/database-index-api.ts

```diff
@@ -79,6 +80,14 @@ export class DatabaseIndexApi<DateType = Date | number> {
    * @param models
    */
   async indexModels(models: Array<IndexModelArgs>): Promise<void> {
+    for (const modelArgs of this.previouslyIndexedModels) {
+      const modelPreviouslyIndexed = this.previouslyIndexedModels.some(function (streamId) {
+        return String(streamId) === String(modelArgs)
+      })
+      if (modelPreviouslyIndexed){
```

Comment:
Can you open a new ticket in HDS to 'Handle model history sync after re-indexing` and add `TODO` as part of a follow up. Thx!